### PR TITLE
QE: Removing ruby code indicated as unused

### PR DESCRIPTION
**Pull request by elariekerboull**
## What does this PR change?

Removes code marked as unused by simplecov tool.

Some additional code might be deleted and some code might be restored following discussions.

## Documentation
- No documentation needed: only internal and user invisible changes

- [X] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/18115
Tracks 
4.2
4.3

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: testsuite/features/step_definitions/common_steps.rb [hunk 1]
```diff
@@ -5,11 +5,6 @@
 require 'securerandom'
 require 'pathname'
 
-When(/^I save a screenshot as "([^"]+)"$/) do |filename|
```
Comment (elariekerboull) [inline testsuite/features/step_definitions/common_steps.rb:8]: This step is never referenced but is probably convenient for debugging purposes.

File: testsuite/features/step_definitions/navigation_steps.rb [hunk 2]
```diff
@@ -880,14 +842,6 @@
   row.set(true)
 end
 
-When(/^I uncheck "([^"]*)" in the list$/) do |text|
-  top_level_xpath_query = "//div[@class=\"table-responsive\"]/table/tbody/tr[.//td[contains(.,'#{text}')]]//input[@type='checkbox']"
-  row = find(:xpath, top_level_xpath_query, match: :first)
-  raise "xpath: #{top_level_xpath_query} not found" if row.nil?
-
-  row.set(false)
-end
```
Comment (elariekerboull) [inline testsuite/features/step_definitions/navigation_steps.rb:889]: This step is only used by the step "I uncheck the * client" line 450 which is never called.

File: testsuite/features/support/cobbler_test.rb [hunk 3]
```diff
@@ -36,16 +30,6 @@ def running?
...
-  def get_list(what)
-    result = []
-    unless %w[systems profiles distros].include?(what)
-      raise "unknown get_list parameter '#{what}'"
-    end
-    ret = @server.call('get_' + what)
-    ret.each { |a| result << a['name'] }
-    result
-  end
-
```
Comment (elariekerboull) [inline testsuite/features/support/cobbler_test.rb:48]: Only referred by commented code line 11

File: testsuite/features/support/custom_formatter.rb [hunk 4]
```diff
@@ -1,6 +1,8 @@
 # Copyright (c) 2019 SUSE LLC
 # Licensed under the terms of the MIT license.
 
+### This file is not indicated as covered however the coverage might be implicit?
+
```
Comment (elariekerboull) [inline testsuite/features/support/custom_formatter.rb:5]: Not sure if the code is actually unused or if the tool failed to consider it.

File: testsuite/features/support/namespaces/api.rb [hunk 5]
```diff
@@ -1,6 +1,8 @@
 # Copyright (c) 2022 SUSE LLC.
 # Licensed under the terms of the MIT license.
 
+### This file has no covered code aside from "initialize(api_test)" and could be deleted otherwise
+
```
Comment (elariekerboull) [inline testsuite/features/support/namespaces/api.rb:5]: Whole file except for the `initialize` method is marked as unused

File: testsuite/features/step_definitions/common_steps.rb [hunk 6]
```diff
@@ -5,11 +5,6 @@
...
-  save_screenshot(filename)
-  attach File.open(filename, 'rb'), 'image/png'
-end
-
```
Comment (srbarrios) [inline testsuite/features/step_definitions/common_steps.rb:12]: Let's keep that one please!
I use that from time to time while debugging scenarios.
You can add a comment on top of it about using this only for debugging or so 👼🏼

File: testsuite/features/support/cobbler_test.rb (hunk 3 above)
Comment (srbarrios) [reply testsuite/features/support/cobbler_test.rb:48]: YEah, fine to remove it.

File: testsuite/features/support/cobbler_test.rb [hunk 7]
```diff
@@ -4,12 +4,6 @@
...
 require 'pp'
 
-# ct = CobblerTest.new( "taylor.suse.de" )
-# bool = ct.running?()
-# bool = ct.system_exists( "bla" )
-# any  = ct.system_get_key( "vbox-ug", "uid" )
-# list = ct.get_list( "systems" )
-
 # Class for clobber test
 class CobblerTest
```
Comment (srbarrios) [inline testsuite/features/support/cobbler_test.rb:8]: This methos here, even if not used, are the ones available in the interface for Cobbler, as a facade component, I think it's fine to keep all of them. Same for other namespaces.

File: testsuite/features/support/custom_formatter.rb (hunk 4 above)
Comment (srbarrios) [reply testsuite/features/support/custom_formatter.rb:5]: We are using it from outside, from the Rakefile.

```
      cucumber_opts = "#{include_profiles} #{include_tags} -f html -o output_#{timestamp}-#{run_set}-$TEST_ENV_NUMBER.html -f json -o output_#{timestamp}-#{run_set}-$TEST_ENV_NUMBER.json #{junit_results} -f CustomFormatter::PrettyFormatter -r features "
```

File: testsuite/features/support/lavanda.rb [hunk 8]
```diff
@@ -44,21 +44,11 @@ def full_hostname
     @in_full_hostname
   end
 
-  def private_ip
-    raise 'empty private_ip, something wrong' if @in_private_ip.empty?
-    @in_private_ip
-  end
-
```
Comment (srbarrios) [inline testsuite/features/support/lavanda.rb:51]: Keep it please. Let's only remove unused step definitions, but let's keep what we offer as part of our modules.

File: testsuite/features/support/lavanda.rb [hunk 9]
```diff
@@ -44,21 +44,11 @@ def full_hostname
...
   def public_ip
     raise 'empty public_ip, something wrong' if @in_public_ip.empty?
     @in_public_ip
   end
 
-  def private_interface
-    raise 'empty private_interface, something wrong' if @in_private_interface.empty?
-    @in_private_interface
-  end
-
```
Comment (srbarrios) [inline testsuite/features/support/lavanda.rb:61]: Keep it please. Let's only remove unused step definitions, but let's keep what we offer as part of our modules.

**Review by srbarrios — CHANGES_REQUESTED** [review]
Let's only consider removal of step definitions please.
From there, I have no objections.

But from the content in support folder, even if we don't use now some of the methods offered by the XMLRPC/HTTP API, I prefer to keep there all available options. Same for the Lavanda module.

**Review by srbarrios — APPROVED** [review]

**Pull request merged**